static/app/components/diamondStatus.tsx

```diff
@@ -0,0 +1,52 @@
+import styled from '@emotion/styled';
+
+import {IconDiamond} from 'sentry/icons';
+import {SVGIconProps} from 'sentry/icons/svgIcon';
+import {ColorOrAlias} from 'sentry/utils/theme';
+
+interface DiamondStatusProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * Color of the diamond
+   */
+  color: ColorOrAlias;
+  /**
+   * Icon component to render inside of the diamond
+   */
+  icon: React.ComponentType<SVGIconProps>;
+}
+
+/**
+ * A status indicator that renders a icon within a diamond
+ */
+function DiamondStatus({color, icon: Icon, ...props}: DiamondStatusProps) {
+  return (
+    <StatusWrapper role="presentation" color={color} {...props}>
+      <DiamondBackground color={color} />
+      <Icon color="white" />
+    </StatusWrapper>
+  );
+}
+
+export {DiamondStatus};
```

Comment:
would it be better to have the export near the function, for instance, `export function DiamondStatus`?

Reply:
we should decide on some standard here, that's a good question!